music21/stream/tools.py

```diff
@@ -0,0 +1,170 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+# Name:         stream/tools.py
+# Purpose:      Additional tools for working with streams
+#
+# Authors:      Mark Gotham
+#
+# Copyright:    Copyright © 2022 Michael Scott Asato Cuthbert
+# License:      BSD, see license.txt
+# ------------------------------------------------------------------------------
+
+from __future__ import annotations
+
+from music21 import clef
+from music21 import environment
+from music21 import key
+from music21 import meter
+from music21 import stream
+
+environLocal = environment.Environment('stream.tools')
+
+
+# ------------------------------------------------------------------------------
+
+def removeDuplicates(thisStream: stream.Stream,
+                     classesToRemove: list | None = None,
```

Comment:
Most operations on streams allow an inPlace argument and add a derivation, so I'd suggest adding that here (since it defaults to operating inPlace.)